safe-global/safe-wallet-web
# feat: Human-readable v2

Depends on https://github.com/safe-global/safe-client-gateway/pull/671

## What it solves

Resolves #2449 

## ToDos

- [x] Add `HumanDescription` component to parse structured data response from CGW once https://github.com/safe-global/safe-client-gateway/pull/671 is merged
- [x] Check usage of `useTransactionType`, `TxType` and `TxInfo` and make sure they are still needed

## How this PR fixes it

- Adjusts the history and queue layout to make room for longer descriptions by combining the type and info column
- Removes now unused components `TxType` and `TxInfo` as well as the `useTransactionType` hook
- Adjusts fallback descriptions as outlined [here](https://www.notion.so/safe-global/UI-integration-9f437dd1d21744518fe79715607c7be7)

## How to test

The changes in this PR affect the history, queue, pending txs in the dashboard and the confirm tx screen

1. Open `/history` of a safe
2. Observe that there are only 4 columns now (nonce, description, date, status)
3. Transfers are now in the format of `Send/Receive <amount> to/from <address/name>`
4. Human Descriptions are more styled now e.g. ERC20 approvals or withdraws will highlighted amounts
5. Multisends are now in the format of `Call multiSend <on contract name><via safe app name> with <x amount> actions` while the existance of contract name and safe app name is optional so it could also be `Call multiSend with <x amount> actions`
6. If the method name of a transaction is known the description has the following format: `Call <method name> on <name/address>`
7. If no method name is known the description has the following format: `Interact with <contract address>`

## Screenshots
<img width="1256" alt="Screenshot 2023-09-08 at 13 24 57" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/7ce95d66-1465-4771-a4ad-05cc4ab0846b">
<img width="1250" alt="Screenshot 2023-09-08 at 13 25 13" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/057c00d7-7b23-473b-b88e-6e6f7970a017">
<img width="1257" alt="Screenshot 2023-09-08 at 13 25 28" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/106720ff-ec6b-4b14-a3e3-3fafa993c49f">
<img width="1255" alt="Screenshot 2023-09-08 at 13 25 59" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/2e18ec4f-a7ed-40d4-bffe-c8e4f28c8990">


## Checklist
